source/extensions/geoip_providers/maxmind/geoip_provider.cc

```diff
@@ -111,6 +126,62 @@ MaxmindDbPtr GeoipProvider::initMaxMindDb(const absl::optional<std::string>& db_
   }
 }
 
+CountryMappingPtr
+GeoipProvider::initCountryMapping(const absl::optional<std::string>& country_mapping_path) {
+  auto country_mapping = std::make_unique<absl::flat_hash_map<std::string, std::string>>();
+  if (country_mapping_path) {
+    if (file_system_.fileExists(country_mapping_path.value())) {
+      auto file_or_error = file_system_.fileReadToEnd(country_mapping_path.value());
+
+      if (!file_or_error.status().ok()) {
+        throwEnvoyExceptionOrPanic(fmt::format("Unable to open country mapping file {}. Error {}",
+                                               country_mapping_path.value(),
+                                               std::string(file_or_error.status().message())));
+      }
+      parseCountryMapping(file_or_error.value(), country_mapping_path.value(), country_mapping);
+    } else {
+      config_->incCountryMappingFileParseError();
+      throwEnvoyExceptionOrPanic(fmt::format(
+          "Country Mapping configuration file {} does not exist.", country_mapping_path.value()));
+    }
+  }
+  return country_mapping;
+}
+
+void GeoipProvider::parseCountryMapping(const std::string& file_data, const std::string& file_path,
+                                        const CountryMappingPtr& country_mapping) {
+  TRY_NEEDS_AUDIT {
+    // Remove lines with comments.
+    envoy::type::matcher::v3::RegexMatcher matcher;
+    *matcher.mutable_google_re2() = envoy::type::matcher::v3::RegexMatcher::GoogleRE2();
+    matcher.set_regex("#.*");
+    auto compiled_regex = Regex::Utility::parseRegex(matcher);
+    std::string content_without_comments = compiled_regex->replaceAll(file_data, "");
+    // Load as json and fill country_mapping_.
+    Json::ObjectSharedPtr json = Json::Factory::loadFromString(content_without_comments);
+    json->iterate([&](const std::string& key, const Json::Object& value) {
+      country_mapping->insert(std::make_pair(key, value.asString()));
+      return true;
+    });
+  }
+  END_TRY
+  catch (Json::Exception&) {
```

Comment:
I think `END_TRY CATCH` might be the expected thing here, in support of a build with disabled exceptions. ([example](https://sourcegraph.com/github.com/envoyproxy/envoy@4994e59af25898d56f496fece3ed24dacbdfa87b/-/blob/source/common/router/header_parser_utils.cc?L54))

Also, including the message from the exception in the rethrown exception would probably be helpful to the user, rather than just telling them the entire file is wrong.